test/index.test.js

```diff
@@ -21,6 +21,21 @@ describe('plugin', () => {
   const server = new Hapi.Server();
 
   server.connection({ port: 80 });
+
+  server.register([
+    require('inject-then'),
+    require('./authentication'),
+    {
+      register: require('../lib'),
+      options: {
+        defaultRate: () => defaultRate,
+        redisClient,
+        requestAPIKey: (request) => request.auth.credentials.api_key,
+        overLimitError: createBoomError('RateLimitExceeded', 429, (rate) => `Rate limit exceeded. Please wait ${rate.window} seconds and try your request again.`)
```

Comment:
that's how it was before, but @robinjoseph08 wanted a test that's aligned more with our implementation